src/components/dataset/DatasetScreen.js

```diff
@@ -87,11 +103,87 @@ class DatasetScreen extends Component {
     dispatchGetDataset({ id });
   }
 
+  componentDidUpdate({ isTabular: prevIsTabular }) {
+    const { isTabular } = this.props;
+
+    if (isTabular !== prevIsTabular) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
+        format: isTabular ? FILE_FORMATS.CSV : FILE_FORMATS.JSON,
+      });
+    }
+  }
+
   componentWillUnmount() {
     const { dispatchClearDataset } = this.props;
     dispatchClearDataset();
   }
 
+  renderFormatButtons = () => {
+    const { format } = this.state;
+    return (
+      <ButtonGroup size="small">
+        <Button
+          variant={format === FILE_FORMATS.JSON ? 'outlined' : 'contained'}
+          color="primary"
+          onClick={() => {
+            this.setState({ format: FILE_FORMATS.JSON });
+          }}
+        >
+          JSON
+        </Button>
+        <Button
+          variant={format === FILE_FORMATS.CSV ? 'outlined' : 'contained'}
+          color="primary"
+          onClick={() => {
+            this.setState({ format: FILE_FORMATS.CSV });
+          }}
+        >
+          CSV
+        </Button>
+      </ButtonGroup>
+    );
+  };
+
+  renderTabularView = () => {
+    const { datasetContent, t, isTabular } = this.props;
+
+    if (!isTabular) {
+      return (
+        <Alert severity="error">
+          {t("The dataset can't be displayed as a csv.")}
```

Comment:
Maybe I would just disable to table view button if the dataset cannot be displayed as a table.